account_controllers/controllers.js

```diff
@@ -176,20 +223,111 @@ module.exports = {
+        }
+
+        const { data: garagesData, error: garagesError } = await supabase
           .from('garages')
-          .update({ operation_hours })
+          .select('operation_hours')
           .eq('user_id', id)
           .single();
-        if (error) {
-          console.error('Error updating operation hours:', error);
+
+        if (garagesError) {
+          console.error(
+            'Error fetching current operation hours:',
+            garagesError
+          );
+          return res.sendStatus(500);
+        }
+
+        const currentOperationHours = garagesData
+          ? garagesData.operation_hours
+          : null;
+
+        if (!currentOperationHours) {
+          return res.status(400).send('Current operation hours not found');
+        }
+
+        const [currentStartHour, currentEndHour] = currentOperationHours
+          .split('-')
+          .map(Number);
+
+        const today = DateTime.local().startOf('day');
+        const next7Days = [...Array(7)].map((_, index) =>
+          today.plus({ days: index })
+        );
+        const formattedNext7Days = next7Days.map((day) =>
+          day.toFormat('MM-dd-yy')
+        );
+
+        const {
+          data: startHourReservations,
+          error: startHourReservationsErrors,
+        } = await supabase
+          .from('reservations')
+          .select('*')
+          .in('status', ['reserved', 'checked-in'])
+          .in('date', formattedNext7Days)
+          .gte('time', currentStartHour)
+          .lt('time', startHour);
+
+        if (startHourReservationsErrors) {
+          console.error(
+            'Error fetching reservations:',
+            startHourReservationsErrors
+          );
           return res.sendStatus(500);
         }
+
+        const { data: endHourReservations, error: endHourReservationsErrors } =
+          await supabase
+            .from('reservations')
+            .select('*')
+            .in('status', ['reserved', 'checked-in'])
+            .in('date', formattedNext7Days)
+            .lte('time', currentEndHour)
+            .gt('time', endHour);
+
+        if (endHourReservationsErrors) {
+          console.error(
+            'Error fetching reservations:',
+            endHourReservationsErrors
+          );
+          return res.sendStatus(500);
+        }
+        if (
+          (startHourReservations && startHourReservations.length > 0) ||
+          (endHourReservations && endHourReservations.length > 0)
+        ) {
+          return res
+            .status(400)
+            .send('Cannot update operation hours due to existing reservations');
+        }
+
+        const { error: updateError } = await supabase
+          .from('garages')
+          .update({ operation_hours })
+          .eq('user_id', id);
+
+        if (updateError) {
+          console.error('Error updating operation hours:', updateError);
+          return res.sendStatus(500);
+        }
+
```

Comment:
That is a beefy query..

Reply:
yep, it was pretty complicated to think about and I couldn't figure out how to chain multiple conditional checks into one query, so right now I have it split the query for subset of the data for comparison. However, if I have to refactor, I will research if its possible to do multiple chaining on supabase querys.